equistore-core/src/c_api/blocks.rs

```diff
@@ -186,25 +186,25 @@ pub unsafe extern fn eqs_block_labels(
         }
 
         let values_gradients = CStr::from_ptr(values_gradients).to_str().unwrap();
-        let basic_block = match values_gradients {
-            "values" => (*block).values(),
+        let block = match values_gradients {
+            "values" => (*block).get_self(),
```

Comment:
Using a reference does not work here?

```suggestion
            "values" => &(*block),
```

Reply:
Well, it didn't work yesterday but it does now... Surely I'd messed something up. I'll change it

Reply:
Maybe it was for the mutable reference? mutable and immutable references have slightly different rules for re-borrowing them. Can you also try this for the mutable one & we'll figure out a solution if rustc is not happy 🦀

Reply:
Oh, you did that already =)

Reply:
I don't know what happened because yesterday the mutable one was not working so I can definitely say that I have no idea what I'm doing